src/main/python/deep_qa/layers/entailment_models/tuple_match.py

```diff
@@ -0,0 +1,156 @@
+from keras import backend as K
+from keras import initializations, activations
+from keras.layers import Layer
+
+from overrides import overrides
+from ...tensors.backend import switch, apply_feed_forward
+
+
+class TupleMatch(Layer):
+    r"""
+    This layer takes as input two tensors cprresponding to two tuples, an answer tuple and a background tuple,
+    and calculates the degree to which the background tuple `entails` the answer tuple.  Entailment is
+    determined by generating a set of entailment features from the tuples (the number of
+    entailment_features = number of tuple slots), and then passing these features into a shallow NN to get an
+    entailment score.
+    Each entailment feature is currently made by comparing the corresponding slots in the two tuples and
+    determining the degree of lexical overlap using the formula:
+        :math:`normalized overlap_s = \dfrac{|A_s \cap B_s|}{|A_s|}`
+    where :math:`s` is the index of the slot, :math:`A_s` is answer tuple slot :math:`s` and :math:`B_s` is
+    background tuple slot :math:`s`.
+
+    Inputs:
+        - tuple_1_input (the answer tuple), shape ``(batch size, num_slots, num_slot_words_t1)``,
+          any mask is ignored.  Here num_slot_words_t1 is the maximum number of words in each of the
+          slots in tuple_1.
+        - tuple_2_input (the background_tuple), shape ``(batch size, num_slots, num_slot_words_t2)``,
+          and again, any corresponding mask is ignored. As above, num_slot_words_t2 is the
+          maximum number of words in each of the slots in tuple_2. This need not match tuple 1.
+
+    Output:
+        - entailment score, shape ``(batch, 1)``
+
+    Parameters
+    ----------
+    - num_hidden_layers : int, default=1
+        Number of hidden layers in the shallow NN.
+    - hidden_layer_width : int, default=4
+        The number of nodes in each of the NN hidden layers.
+    - initialization : string, default='glorot_uniform'
+        The initialization of the NN weights
+    - hidden_layer_activation : string, default='relu'
+        The activation of the NN hidden layers
+    - final_activation : string, default='sigmoid'
+        The activation of the NN output layer
+    """
+
+    def __init__(self, num_hidden_layers: int=1, hidden_layer_width: int=4,
+                 initialization: str='glorot_uniform', hidden_layer_activation: str='relu',
+                 final_activation: str='sigmoid'):
+        """
```

Comment:
This docstring isn't really adding anything, and the parameters are covered in the class docstring anyway.  I'd probably remove this.